.github/workflows/backend-build.yml

```diff
@@ -0,0 +1,83 @@
+name: Backend Build Tests
+
+on:
+  push:
+    branches:
+      - main
+      - master
+      - gh-action-build-backend # TODO delete this if ever merged to master/main
+  pull_request:
+    branches:
+      - main
+      - master
+      - gh-action-build-backend # TODO delete this if ever merged to master/main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.17.0'
+
+    - name: Install dependencies
+      run: |
+        go version
+        go get -u golang.org/x/lint/golint
+
+  iptables:
+    name: build backend package iptables
+    needs: setup
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    
+    - name: Build backends/iptables
+      run: |
+        cd ./backends/iptables
+        go mod vendor
+        go build
+
+  ipvs:
+    name: build backend package ipvs
+    needs: setup
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    
+    - name: Build backends/ipvs
+      run: |
+        cd ./backends/ipvs
+        go mod vendor
+        go build
+  
+  ipvs-as-sink:
+    name: build backend package ipvs-as-sink
+    needs: setup
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    
+    - name: Build backends/ipvs-as-sink
+      run: |
+        cd ./backends/ipvs-as-sink
+        go mod vendor
+        go build
+  
+  nft:
+    name: build backend package nft
+    needs: setup
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    
+    - name: Build backends/nft
+      run: |
+        cd ./backends/nft
+        go mod vendor
+        go build
```

Comment:
imo this is a good start !

Reply:
wouldn't `go mod download` be enough? Sorry I tend to be a semantic nazi :p